test/upgrade/installation/git_head.go

```diff
@@ -0,0 +1,40 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package installation
+
+import (
+	pkgupgrade "knative.dev/pkg/test/upgrade"
+)
+
+func GitHead() pkgupgrade.Operation {
+	return pkgupgrade.NewOperation("EventingGitHead", func(c pkgupgrade.Context) {
+		ops := []string{
+			"install_head",
+			"install_channel_crds",
+			"install_mt_broker",
+			"install_sugar",
+		}
```

Comment:
I am a little bit concerned about this as sometimes there may be hacky code in bash script to install it properly, eg waiting to make sure all the pods in the previous step are running before applying next. I am wondering if there is a bash function that does all these installations and we can just call the bash function instead. But for sure we can change it later if such a bash function is required.

Reply:
There are no such function. Each of those functions has appropriate waits built inside, so it isn't needed.

Current code executes them directly: https://github.com/knative/eventing/pull/4519/files#diff-1c4d417537c9878a5068005f79761334318400204068c4dfd4a7656226bfefe1L61-L64